### ISSUE TITLE
Training on Intermediate Logits with Cross-Encoder

### ISSUE THREAD

**Issue by andreawynn** [UKPLab/sentence-transformers#3435]
I am trying to use the CrossEncoderTrainer class for model distillation with the Qwen3-Reranker-0.6B-seq-cls model, where I would need to access the intermediate logits (i.e. the output of the final layer before the classifier). Is this currently supported?

**Comment by tomaarsen** [comment]
Hello!

So I understand correctly: you'd like to perform distillation on the hidden states (i.e. the classifier inputs) instead of the model outputs (i.e. the classifier outputs)?

The usual approach is to perform distillation with the full teacher on the full student.

But your approach is also possible, just not fully out of the box. In short, you'll likely have to:
1. Subclass the `CrossEncoder` and override the `__init__` with almost the same code, but using `AutoModel` instead of `AutoModelForSequenceClassification`.
2. Use a custom loss function: the prebuilt losses expect that the model outputs a class logit. You can probably use a loss that's very similar to MSELoss (from the Sentence Transformers losses, not the Cross Encoder one), comparing your training model predictions with the "labels".
3. For the teacher model, grab a training dataset (e.g. a dataset with 2 columns for query and answer), and add a column for hidden states from the teacher model.

Then you can train the underlying encoder to match the teacher model, but the classifier head is still untrained. Likely you'll have to save your "headless" model, and then load it with the normal CrossEncoder (so it adds a head with random weights). You'll then have to train that model like normal, although you can also choose to freeze the weights of the encoder, so that only the head will be trained.

I hope this gives you an impression of how it would be possible, then you can make a choice of if you want to give it a try.

- Tom Aarsen
